docs/style-guide.md

```diff
@@ -301,6 +433,80 @@ let x = match number {
 
 Obviously where an if-statement is simply there to do optional work, and not cover every case, it may well be more suitable to just use that.
 
+# Constructors :: suggested
+
+If the struct doesn't require any arguments to be initialied, it should implement or derive the `Default` trait.
+For structs with argumented constructors, `new` or `try_new` methods should be used for canonical construction, `From` and `TryFrom` traits should be implemented for generic conversions and `from_*` and `try_from_*` for non-trait based specific conversions.
+
+| Type | Fallible | Trait | Use |
+| ---- | ---- | ---- | ---- |
+| Default | 𐄂 | `Default` | `Struct::default();` |
+| Arguments | 𐄂 | | `Struct::new(locale, options);` |
+| Arguments | ✓ | | `Struct::try_new(locale, options)?;` |
+| `&str` | ✓ | `FromStr` | `value.parse()?` |
+| Type | 𐄂 | `From` | `Struct::from(value);` |
+| Type | ✓ | `TryFrom` | `Struct::try_from(value)?;` |
+| Other | 𐄂 | | `Struct::from_{name}(value);` |
+| Other | ✓ | | `Struct::try_from_{name}(value)?;` |
+
+## Options
+
+Many ICU related constructors require a number of options to be passed. In such cases, it is recommended to provide a separate structure that is used to assemble all required options, and pass it to the constructor.
+
+### Examples
+
+```rust
+#[derive(Default)]
+struct MyStructOptions {
+    pub min_fraction_digits: usize,
+    pub max_fraction_digits: usize,
+}
+
+impl Default for MyStructOptions {
+    pub fn default() -> Self {
+        Self {
+            min_fraction_digits: 3,
+            max_fraction_digits: 5,
+        }
+    }
+}
+
+impl MyStructOptions {
+   // Additional helper methods for setting the options,
+   // or validating the consistency of the options.
+}
+
+struct MyStruct {
+  locale: Locale,
+  fraction_digits: Range<usize>,
+}
+
+impl MyStruct {
+    pub fn new(locale: Locale, options: MyStructOptions) -> Result<Self, ()> { ... };
+}
+
+fn main() {
+    let mut options = MyStructOptions::default();
+    options_max_fraction_digits = 10;
+    // Optional debug time validation of the options
+    assert!(options.validate());
+
+    let s = MyStruct::try_new(locale, options).expect("Construction failed.");
+}
+```
+
+It is also recommended that such structs implement `Default` trait to simplify common construction models:
```

Comment:
"recommended" is too weak.  What are cases where you *wouldn't* want to implement `Default`?  If there are none, I would strengthen this to "required", and we can revisit this requirement if antipattern cases come up.